components/details/details.controller.js

```diff
@@ -0,0 +1,24 @@
+const detailsController = function ($scope, productsService, $routeParams, $location) {
+
+  this.productsArray = [];
+  this.selectedProduct = [];
+
+  this.getSelectedProduct = async () => {
+    const product = await productsService.getProducts();
+    this.productsArray = product.products;
+    this.selectedProduct = this.productsArray.filter(ele => ele.id === +$routeParams.id);
```

Comment:
what's that "+"? `... +$routeParams.id`

Reply:
I thought I should put a + to specify that the param "id" is a number, not a string from the url. I guess it's not necessary because of non-strong typed language?

Reply:
Actually, it doesn't work without the "+" because i used ===; i could change everything just in == without the "+"

Reply:
or the opposite ... https://www.w3schools.com/jsref/jsref_parseint.asp